File: C++/Assignment 1/problem2.cpp
#include <iostream> 
using namespace std;

class Batsman
{
    private:
        int bcode;
        char bname[20];
        int innings, notout, runs; 
        int batavg;
        void calcavg()
        {
            batavg = runs / (innings - notout);
        }

    public:
        void readddata(); 
        void displaydata();
};

void Batsman::readddata()
{
    cout << "Enter batsman code: "; 
    cin >> bcode;
    while (bcode < 999 || bcode > 10000)
    {
        cout << "Please enter a correct four digit batsman code: "; 
        cin >> bcode;
    }
    cout << "Enter batsman name: "; 
    cin.ignore(1000, '\n'); 
    cin.getline(bname, sizeof(bname)); 
    cout << "Enter amount of innings: "; 
    cin >> innings;
    cout << "Enter amount of notouts: "; 
    cin >> notout;
    cout << "Enter amount of runs: "; 
    cin >> runs;
    calcavg();
}

void Batsman::displaydata()
{
    cout << "\nBatsman code: " << bcode << "\nBatsman name: " << bname
            << "\nInnings: " << innings << "\nNotouts:" << notout
            << "\nRuns: " << runs << "\nBatting average: " << batavg;
}

int main()
{
    Batsman b1; 
    b1.readddata(); 
    b1.displaydata(); 
    return 0;
}

File: C#/Test/problem2.cs
using System;
using System.ComponentModel;
using System.Runtime.ExceptionServices;

public class TestProblem2
{
    class Batsman
    {
        private int bcode;
        private char[] bname = new char[20]; 
        private int innings, notout, runs; 
        private float batavg;
        private void calcavg()
        {
        batavg = (float)runs / (innings - notout);
        }

        public void readdata()
        {
            Console.WriteLine("Enter batsman code:"); 
            bcode = int.Parse(Console.ReadLine()); 
            while (bcode < 999 || bcode > 10000)
            {
                Console.WriteLine("Please enter a correct four digit batsman code: ");
                bcode = int.Parse(Console.ReadLine());
            }

            Console.WriteLine("Enter batsman name:"); 
            string nameString = Console.ReadLine(); 
            for (int i = 0; i < nameString.Length; i++)
            {
                bname[i] = nameString[i];
            }

            Console.WriteLine("Enter amount of innings: "); 
            innings = int.Parse(Console.ReadLine()); 
            Console.WriteLine("Enter amount of notouts: "); 
            notout = int.Parse(Console.ReadLine()); 
            Console.WriteLine("Enter amount of runs:"); 
            runs = int.Parse(Console.ReadLine()); 
            calcavg();
        }

        public void displaydata()
        {
            string nameString = new string(bname); 
            Console.WriteLine("\nBatsman code: " + bcode + "\nBatsman name: "
                + nameString + "\nNumber of Innings: " + innings 
                + "\nNumber of Notouts: " + notout + "\nNumber of Runs: " + runs 
                + "\nBatting Average: " + batavg.ToString("0.00"));
        }
    }

    public static void Main(string[] args)
    {
        Batsman b = new Batsman(); 
        b.readdata(); 
        b.displaydata();
    }
}